src/core/storage/sqliteStorage/utils.ts

```diff
@@ -14,22 +16,33 @@ function getUnMigrationSqls(
   if (versionCompare(latestVersion, currentVersion) == 0) {
     return null;
   }
-  const sqlUnMigrations: string[] = [];
-  MIGRATIONS.forEach((migration) => {
+
+  const migrationStatements: { statement: string; values?: unknown[] }[] = [];
+  for (const migration of MIGRATIONS) {
     if (versionCompare(currentVersion, migration.version) == -1) {
-      sqlUnMigrations.push(...migration.sql);
+      if (migration?.sql) {
+        migration?.sql.forEach((sql) => {
+          migrationStatements.push({
+            statement: sql,
+          });
+        });
+      } else if (migration.migrationStatements) {
+        const statements = await migration.migrationStatements(session);
+        migrationStatements.push(...statements);
```

Comment:
several places in this PR we are using `...` to clone, can we clean that up and e.g. here just have `.push(statements)`?

Reply:
`statements` is array, Do you mean ` migrationStatements.concat(statements)`?

Reply:
Oh, sorry makes sense now. So we can keep it. (though I think concat is more readable)